emcc.py

```diff
@@ -1387,8 +1387,8 @@ def default_setting(name, new_default):
       shared.Settings.MIN_EDGE_VERSION = 0
       shared.Settings.MIN_CHROME_VERSION = 0
 
-    if shared.Settings.MIN_SAFARI_VERSION <= 9 and shared.Settings.WASM2JS:
-      shared.Settings.WORKAROUND_IOS_9_RIGHT_SHIFT_BUG = 1
+    if shared.Settings.MIN_SAFARI_VERSION <= 90305:
+      exit_with_error('MIN_SAFARI_VERSION <= 90305: Emscripten does not support iPhone 4s, iPad 2, iPad 3, iPad Mini 1, Pod Touch 5 (devices with end-of-life at iOS 9.3.5) and older. (browser has no Wasm support, and Wasm2JS support would not run). Please pass -s MIN_SAFARI_VERSION=100000 or higher.')
```

Comment:
There is an `if shared.Settings.LEGACY_VM_SUPPORT:` in this file that then sets `MIN_SAFARI_VERSION` to 0. It seems like that would conflict with erroring here, making that flag no longer work.

I think the error is too extreme. How about not erroring, and adding some docs to `settings.js` to see that "0" means 90305 or later? Even if not stated in the others, we surely have other such minimal versions in all browsers.

Reply:
> Even if not stated in the others, we surely have other such minimal versions in all browsers.

Agree. I view this rather that we have just not done the work to identify those minimal versions on other browsers. I think it would be nice to be explicit about those minimal versions in other browsers as well, since that will carry valuable information to users.

However dropped this diagnostic, how does that look now?